LuaRules/Gadgets/ai_simple_eco.lua

```diff
@@ -0,0 +1,690 @@
+				Echo("fusion")
+			else
+				buildOrAssistCloseTo(unitId, singuDefID, teamId, x + 100, y, z - 50, 3000)
+				Echo("singu")
+			end
+		end
+	end
+end
+
+local function newConRoles(unitId, thisTeamData)
+	if not thisTeamData.conRoles[unitId] then
+		if thisTeamData.numMexers < 2 then
+			thisTeamData.conRoles[unitId] = MEXER
+			thisTeamData.numMexers = thisTeamData.numMexers + 1
+		elseif thisTeamData.numEMakers < 2 then
+			thisTeamData.conRoles[unitId] = E_MAKER
+			thisTeamData.numEMakers = thisTeamData.numEMakers + 1
+		elseif thisTeamData.numMexers < 3 then
+			thisTeamData.conRoles[unitId] = MEXER
+			thisTeamData.numMexers = thisTeamData.numMexers + 1
+		elseif thisTeamData.numEMakers < 4 then
+			thisTeamData.conRoles[unitId] = E_MAKER
+			thisTeamData.numEMakers = thisTeamData.numEMakers + 1
+		elseif thisTeamData.numGridders * 6 < thisTeamData.numEMakers - 2 then
+			thisTeamData.conRoles[unitId] = GRIDDER
+			thisTeamData.numGridders = thisTeamData.numGridders + 1
+		else
+			thisTeamData.conRoles[unitId] = E_MAKER
+			thisTeamData.numEMakers = thisTeamData.numEMakers + 1
+		end
+		printThing("thisTeamData.conRoles", thisTeamData.conRoles)
+		local x, y, z = spGetUnitPosition(unitId)
+		--Spring.MarkerAddPoint ( x, y, z, thisTeamData.conRoles[unitId], true)
+	end
+end
+
+local function startAreaConOrders(teamId, unitId)
+	local current, storage, _, income = spGetTeamResources(teamId, "metal")
+	local facs = spGetTeamUnitsByDefs(teamId, cloakfacDefID)
+	local x, y, z = spGetUnitPosition(unitId)
+	local facX, facY, facZ = spGetUnitPosition(facs[1])
+	--Echo("Storage " .. storage)
+	if storage < HIDDEN_STORAGE + 100 and not isXInRange(teamId, facX, facZ, 9000, storageDefID, true)  then
+		local xx = x - 100
+		local zz = z
+		local yy = max(0, spGetGroundHeight(xx, zz))
+
+		buildOrAssistCloseTo(unitId, storageDefID, teamId, xx, yy, zz, 2000)
+		return
+	end
+end
+
+local function newConOrders(teamId, unitId, thisTeamData)
+	startAreaConOrders(teamId, unitId)
+	conRoleOrders(teamId, unitId, thisTeamData)
+end
+
+local function oldConOrders(teamId, cmdQueue, unitId, thisTeamData)
+	local facs = spGetTeamUnitsByDefs(teamId, cloakfacDefID)
+	local x, y, z = spGetUnitPosition(unitId)
+	-- Rebuild fac
+	if #facs == 0 then
+		-- TODO
+		buildOrAssistCloseTo(unitId, cloakfacDefID, teamId, x + 150, y, z + 50, 2000)
+		return
+	end
+
+	local startpos = thisTeamData.startpos
+
+	if sqDistance(x,z, startpos[1], startpos[2]) < 1000000 then
+		--Echo("new con orders")
+		startAreaConOrders(teamId, unitId)
+	end
+
+	-- Always reclaim
+	--if hard then
+	--	spGiveOrderToUnit(unitId, 90, {x, y, z, 300}, {shift=true})
+	--end
+	conRoleOrders(teamId, unitId, thisTeamData)
+end
+
+local function factoryOrders(teamId, unitId, frame)
+	if spGetFactoryCommands(unitId, 0) == 0 then
+		spGiveOrderToUnit(unitId, -conjurerDefID, {}, {})
+		spGiveOrderToUnit(unitId, 115, {1}, {}) -- repeat build
+		spGiveOrderToUnit(unitId, 34220, {0}, {}) -- priority low
```

Comment:
Prefer named order constants over magic numbers, table-less parameters when there's a single one, and numerical command options:
```suggestion
		spGiveOrderToUnit(unitId, -conjurerDefID, 0, 0)
		spGiveOrderToUnit(unitId, CMD.REPEAT, 1, 0) -- repeat build
		spGiveOrderToUnit(unitId, CMD_PRIORITY, 0, 0) -- priority low
```